### PR TITLE
Added chat screens

### PR DESCRIPTION
**Pull request by Deepanshu-Rohilla**
1. Created whole UI of messaging component.
2. Updated the app bar. Now there is separate app bar for different screens.

### CODE REVIEW

**Comment by Deepanshu-Rohilla** [comment]
The background in chat bubble needs more attention.

File: lib/main.dart [hunk 1]
```diff
@@ -32,15 +42,17 @@ final lightTheme = ThemeData(
...
     );
 
 class ThemeNotifier with ChangeNotifier {
   ThemeData _themeData;
 
   ThemeNotifier(this._themeData);
-  getTheme() => _themeData;
+  ThemeData getTheme() => _themeData;
 
+  // ignore: always_declare_return_types
```
Comment (arpit-saxena) [inline lib/main.dart:55]: Why? Just figure out the return type and write it there.

File: lib/screens/chatscreen/chat_components/chat_bubble.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(left: 16, right: 16, top: 10, bottom: 10),
+      child: Align(
+        alignment: (widget.chatMessage.type == MessageType.Receiver ? Alignment
+            .topLeft : Alignment.topRight),
+        child: Container(
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(30),
+            color: (widget.chatMessage.type == MessageType.Receiver ? Colors.grey : Colors.grey),
```
Comment (arpit-saxena) [inline lib/screens/chatscreen/chat_components/chat_bubble.dart:25]: Define the colors in main.dart and use them here. We want all the theming in one place

File: lib/screens/chatscreen/chat_modules/chat_detail_page.dart [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    ChatMessage(message: 'Oh! Nice. Same here man', type: MessageType.Sender),
+    ChatMessage(message: 'Oh! Nice. Same here man', type: MessageType.Sender),
+  ];
+
+  List<SendMenuItems> menuItems = [
+    SendMenuItems(text: 'Photos & Videos', icons: Icons.image, color: Colors.amber),
+    SendMenuItems(text: 'Document', icons: Icons.insert_drive_file, color: Colors.blue),
+    SendMenuItems(text: 'Audio', icons: Icons.music_note, color: Colors.orange),
+    SendMenuItems(text: 'Location', icons: Icons.location_on, color: Colors.green),
+    SendMenuItems(text: 'Contact', icons: Icons.person, color: Colors.purple),
```
Comment (arpit-saxena) [inline lib/screens/chatscreen/chat_modules/chat_detail_page.dart:49]: Define these colors in main.dart. I guess use whatever names for now. @tmibvishal can guide on that

File: lib/screens/chatscreen/chat_modules/chat_detail_page.dart [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  ];
+
+  void showModal(){
+    showModalBottomSheet(
+        context: context,
+        builder: (context){
+          return Container(
+            height: MediaQuery.of(context).size.height/2,
+            color: Color(0xff737373),
```
Comment (arpit-saxena) [inline lib/screens/chatscreen/chat_modules/chat_detail_page.dart:58]: main

File: lib/screens/chatscreen/chat_modules/chat_detail_page.dart [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+                children: <Widget>[
+                  GestureDetector(
+                    onTap: (){
+                      showModal();
+                    },
+                    child: Container(
+                      height: 40,
+                      width: 40,
+                      decoration: BoxDecoration(
+                        color: Colors.blueGrey,
```
Comment (arpit-saxena) [inline lib/screens/chatscreen/chat_modules/chat_detail_page.dart:145]: main

File: lib/screens/chatscreen/chat_modules/chat_page.dart [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          children: <Widget>[
+            Padding(
+              padding: EdgeInsets.only(top: 16,left: 16,right: 16),
+              child: TextField(
+                decoration: InputDecoration(
+                  hintText: 'Search...',
+                  hintStyle: TextStyle(color: Colors.grey.shade400),
+                  prefixIcon: Icon(Icons.search,color: Colors.grey.shade400,size: 20,),
+                  filled: true,
+                  fillColor: Colors.grey.shade100,
```
Comment (arpit-saxena) [inline lib/screens/chatscreen/chat_modules/chat_page.dart:41]: colors in main

File: lib/screens/chatscreen/chat_modules/chat_page.dart [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                  contentPadding: EdgeInsets.all(8),
+                  enabledBorder: OutlineInputBorder(
+                      borderRadius: BorderRadius.circular(30),
+                      borderSide: BorderSide(
+                          color: Colors.grey.shade100
```
Comment (arpit-saxena) [inline lib/screens/chatscreen/chat_modules/chat_page.dart:46]: ..

**Comment by arpit-saxena** [comment]
PS: Fix the linting errors too

File: lib/screens/chatscreen/chat_modules/chat_detail_page.dart (hunk 3 above)
Comment (tmibvishal) [reply lib/screens/chatscreen/chat_modules/chat_detail_page.dart:49]: Make a function in main.dart and pass an int i in it to get that color for menu

**Pull request merged**